rfcs/text/0428-extending-and-duplicating-types.md

```diff
@@ -206,19 +206,95 @@ Furthermore, resolving the intentions of the existing definition for the message
 
 We propose that we should treat type updates as complete replacements, so implementation is much more straightforward for embedding applications. Partial schema updates also add some level of indirection, and that may obfuscate error sources and error reasons. Therefore, updates to types must be complete replacements rather than partial schema updates.
 
+### Removing link ordering
+
+Adding a type extension mechanism causes issues with the behavior of links in situations where a link type extends another link type, and an entity type refers to both of them. This, combined with a few other reasons explored below, has lead to us suggesting the removal of link ordering from the specification.
+
+#### Conflicting Link Orderings
+
+To understand this, let's take the example of `Person` above. Let's define a `Knows` link type, and a `Has Friend` link type that extends `Knows`. Now let's say that `Person` defines `Knows` and `Has Friend` links to `Person` entities:
+
+```json
+{
+  "$id": "https://blockprotocol.org/@alice/entity-type/person/v/1",
+  "title": "Person",
+  // ...
+  "links": {
+    "https://blockprotocol.org/@alice/entity-type/knows/v/1": {
+      "type": "array",
+      "items": {
+        "oneOf": [
+          { "ref": "https://blockprotocol.org/@alice/entity-type/person/v/1" }
+        ]
+      }
+    },
+    "https://blockprotocol.org/@alice/entity-type/has-friend/v/1": {
+      "type": "array",
+      "items": {
+        "oneOf": [
+          { "ref": "https://blockprotocol.org/@alice/entity-type/person/v/1" }
+        ]
+      }
+    }
+  }
+}
+```
+
+Now, let's say that a given `Person`, `Alice` has the following links:
+
+- `KnowsBob`
+- `HasFriendCharlie`
+- `HasFriendDave`
+
+Now, `HasFriendCharlie` and `HasFriendDave` are both link entities of type `Has Friend`, which extends `Knows`, which means they are also `Knows` links. Meaning if you were to query for `Has Friend` links for `Alice` you'd get:
+
+```ts
+getHasFriendLinks(alice); // [ HasFriendCharlie, HasFriendDave ]
+```
+
+and if you were to query for `Knows` links for `Alice` you'd get:
+
+```ts
+getKnowsLinks(alice); // [ KnowsBob, HasFriendCharlie, HasFriendDave ]
+```
+
+Now, what if we said that both of these link collections were **ordered**? We could define an order on the link entities, say
+
+```ts
+HasFriendCharlie.leftToRightOrder = 1;
+HasFriendDave.leftToRightOrder = 0;
+```
+
+Giving us:
+
+```ts
+getHasFriendLinks(alice); // [ HasFriendDave, HasFriendCharlie ]
+```
```

Comment:
A bit of a nit, but we dont have other examples like these in the RFC where we usually just write plain JSON. Any reason why we opted to use pseudeo-code blocks for this instead of writing out explicitly how the data looks?

Reply:
I can imagine that using JSON objects here everywhere will get pretty verbose. cc @Alfred-Mountfield

Reply:
I don't mean that we should use the full JSON objects, but just that these operations are sort of an unknown/new syntax at this point and that we'd have to be a little more explicit about what it means

Reply:
I had written it in the comment in pseudo code as part of an explanation of what I think we needed to add. I didn't necessarily mean we should copy paste it into the RFC 😅 

I agree that having big JSON objects would be far too verbose, so I'm not sure how we go about representing this in a concise manner. Perhaps there is a more intuitive syntax we can use, or perhaps there's a more explainable one as you say @thehabbos007

Reply:
What about plain text? 🙃